app/controllers/dislikes_controller.rb

```diff
@@ -1,16 +1,20 @@
 class DislikesController < ApplicationController
-  before_action :authenticate_user!
   before_action :set_item, only: :create
   before_action :set_dislike, only: :destroy
   before_action :verify_non_repeat, only: :create
 
   def create
-    if @item.dislikes.create!(user_id: current_user.id)
-      flash[:success] = "You have disliked this #{@item.class}"
+    if signed_in?
```

Comment:
I believe you can just leave this file the way it was. Using `before_action :authenticate_user!` will require a user to be signed in to perform this action, and if they are not it will redirect them to the sign in page